Communitter/src/main/java/com/communitter/api/service/InvitationService.java

```diff
@@ -0,0 +1,69 @@
+package com.communitter.api.service;
+
+import com.communitter.api.dto.InvitationDto;
+import com.communitter.api.dto.request.InvitationCreateRequestDto;
+import com.communitter.api.key.SubscriptionKey;
+import com.communitter.api.model.Community;
+import com.communitter.api.model.Invitation;
+import com.communitter.api.model.Role;
+import com.communitter.api.model.Subscription;
+import com.communitter.api.model.User;
+import com.communitter.api.repository.CommunityRepository;
+import com.communitter.api.repository.InvitationRepository;
+import com.communitter.api.repository.RoleRepository;
+import com.communitter.api.repository.SubscriptionRepository;
+import com.communitter.api.repository.UserRepository;
+import com.communitter.api.util.InvitationStatus;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class InvitationService {
+
+    private final CommunityRepository communityRepository;
+    private final InvitationRepository invitationRepository;
+    private final SubscriptionRepository subscriptionRepository;
+    private final RoleRepository roleRepository;
+    private final UserRepository userRepository;
+
+    public InvitationDto inviteUser(User authUser, InvitationCreateRequestDto request) {
+
+        Community community = communityRepository.getReferenceById(request.getCommunityId());
+
+        if (community.isPublic()) {
+            throw new RuntimeException("Cannot invite a user for a public community");
+        }
+
+        User invitedUser = userRepository.findByUsername(request.getUsername()).orElseThrow();
```

Comment:
Can we throw specific messages. Also we can add try catch block to service and handle it in the controller

Reply:
orElseThrow sends "Value not present" message. this is what i want to send actually. There is global exception handling mechanism so we do not need try catch block.